src/stream/src/executor/source/source_executor.rs

```diff
@@ -548,8 +549,10 @@ impl<S: StateStore> SourceExecutor<S> {
                     last_barrier_time = Instant::now();
 
                     if self_paused {
```

Comment:
There are two other places we use `self_paused`:
- [source_backfill_executor](https://github.com/risingwavelabs/risingwave/blob/d856dd9df1bf498860d15ae633a90ec44135d441/src/stream/src/executor/source/source_backfill_executor.rs#L497)
- [fs_source_executor](https://github.com/risingwavelabs/risingwave/blob/d856dd9df1bf498860d15ae633a90ec44135d441/src/stream/src/executor/source/fs_source_executor.rs#L398)

It is better to fix them as well.